File: compatibility-tests/src/test/java/nnl/rocks/kactoos/collection/SyncCollectionTest.java
/**
 * The MIT License (MIT)
 * <p>
 * Copyright (c) 2017-2018 Yegor Bugayenko
 * <p>
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 * <p>
 * The above copyright notice and this permission notice shall be included
 * in all copies or substantial portions of the Software.
 * <p>
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */
package nnl.rocks.kactoos.collection;

import nnl.rocks.kactoos.matchers.RunsInThreads;
import org.hamcrest.MatcherAssert;
import org.junit.Test;

/**
 * Test Case for {@link SyncCollection}.
 * @author Yegor Bugayenko (yegor256@gmail.com)
 * @version $Id: 61032113b5bb1fbb4e21390f307c22fcde3c189f $
 * @since 0.23
 * @checkstyle JavadocMethodCheck (500 lines)
 */
public final class SyncCollectionTest {

    @Test
    public void behavesAsCollection() {
        MatcherAssert.assertThat(
            "Can't behave as a collection",
            new SyncCollection<>(1, 2, 0, -1),
            new BehavesAsCollection<>(-1)
        );
    }

    @Test
    public void worksInThreads() {
        MatcherAssert.assertThat(
            "Can't behave as a collection in multiple threads",
            list -> {
                MatcherAssert.assertThat(list, new BehavesAsCollection<>(0));
                return true;
            },
            new RunsInThreads<>(new SyncCollection<>(1, 0, -1, -1, 2))
        );
    }

}

File: kactoos-jvm/src/test/kotlin/nnl/rocks/kactoos/collection/SyncCollectionTest.kt
package nnl.rocks.kactoos.collection

import nnl.rocks.kactoos.func.FuncOf
import nnl.rocks.kactoos.matchers.RunsInThreads
import org.hamcrest.MatcherAssert
import org.junit.Test

/**
 * Test Case for [SyncCollection].
 * @author Yegor Bugayenko (yegor256@gmail.com)
 * @version $Id: 61032113b5bb1fbb4e21390f307c22fcde3c189f $
 * @since 0.23
 * @checkstyle JavadocMethodCheck (500 lines)
 */
class SyncCollectionTest {

    @Test
    @Throws(Exception::class)
    fun behavesAsCollection() {
        MatcherAssert.assertThat(
            "Can't behave as a collection",
            SyncCollection(1, 2, 0, - 1),
            BehavesAsCollection(- 1)
        )
    }

    @Test
    fun worksInThreads() {
        MatcherAssert.assertThat(
            "Can't behave as a collection in multiple threads",
            FuncOf { list ->
                MatcherAssert.assertThat(list, BehavesAsCollection(0))
                true
            },
            RunsInThreads(SyncCollection(1, 0, - 1, - 1, 2))
        )
    }
}
